compiler/AST/AstToText.cpp

```diff
@@ -1241,48 +1235,29 @@ void AstToText::appendExpr(CallExpr* expr, bool printingType)
         appendExpr(expr, fnName, printingType);
     }
 
-    else if (isSymExpr(expr->baseExpr))
+    else if (isSymExpr(expr->baseExpr) || isCallExpr(expr->baseExpr))
     {
-      if (expr->numActuals() == 0)
-      {
-        // NOAKES 2015/02/05  Debugging support.
-        // Might become ASSERT in the future
-        mText += "AppendExpr.Call06";
-      }
-
-      else if (expr->numActuals() == 1)
+      if (isSymExpr(expr->baseExpr))
       {
         appendExpr(expr->baseExpr, printingType);
-        mText += '(';
-        appendExpr(expr->get(1), printingType);
-        mText += ')';
       }
 
       else
       {
-        appendExpr(expr->baseExpr, printingType);
-        mText += '(';
-
-        for (int i = 1; i <= expr->numActuals(); i++)
-        {
-          if (i > 1)
-            mText += ", ";
-
-          appendExpr(expr->get(i), printingType);
-        }
-
-        mText += ')';
+        CallExpr* subCall = toCallExpr(expr->baseExpr);
+        appendExpr(subCall->get(1), printingType);
+        mText += '.';
+        appendExpr(subCall->get(2), printingType);
+      }
+      
+      if (expr->numActuals() == 0 && isSymExpr(expr->baseExpr))
+      {
+        // NOAKES 2015/02/05  Debugging support.
+        // Might become ASSERT in the future
+        mText += "AppendExpr.Call06";
       }
-    }
-
-    else if (isCallExpr(expr->baseExpr))
-    {
-      CallExpr* subCall = toCallExpr(expr->baseExpr);
-      appendExpr(subCall->get(1), printingType);
-      mText += '.';
-      appendExpr(subCall->get(2), printingType);
 
-      if (expr->numActuals() == 0)
+      else if (expr->numActuals() == 0 && isCallExpr(expr->baseExpr))
```

Comment:
I was asking about the `expr->numActuals() ==` branches, not if SymExpr and CallExprs could be handled by the same branch.  The branch for `expr->numActuals() > 1` should generate the same output in the 0 and 1 case without needing to explicitly have their own if branch.  I don't think having SymExpr and CallExpr handled by the same branch is appropriate

Reply:
Sorry for the misunderstanding. I'll revert the change and have it that way.